server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java

```diff
@@ -0,0 +1,270 @@
+
+            channel.exchangeDeclare(EXCHANGE_NAME, "direct", DURABLE);
+            channel.queueDeclare(QUEUE, DURABLE, !EXCLUSIVE, !AUTO_DELETE, ImmutableMap.of()).getQueue();
+            channel.queueBind(QUEUE, EXCHANGE_NAME, ROUTING_KEY);
+
+            for (int i = 0; i < MESSAGE_COUNT; i++) {
+                channel.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPS, String.valueOf(i).getBytes(StandardCharsets.UTF_8));
+            }
+
+            channel.basicConsume(QUEUE, callback(result, channel));
+            Thread.sleep(ONE_MINUTE);
+        }
+
+        assertThat(result)
+            .containsOnlyElementsOf(
+                IntStream.range(0, MESSAGE_COUNT)
+                    .boxed()
+                    .collect(Guavate.toImmutableList()));
+    }
+
+    @Test
+    public void nodeKillingWhenConsuming(DockerRabbitMQCluster cluster) throws Exception {
+        ConcurrentLinkedDeque<Integer> result = new ConcurrentLinkedDeque<>();
+
+        ConnectionFactory connectionFactory1 = RabbitMQUtils.prepareConnectionFactory(cluster.getRabbitMQ1());
+        ConnectionFactory connectionFactory2 = RabbitMQUtils.prepareConnectionFactory(cluster.getRabbitMQ2());
+
+        ImmutableList<Address> addresses = ImmutableList.of(
+            new Address(cluster.getRabbitMQ1().getHostIp(), cluster.getRabbitMQ1().getPort()),
+            new Address(cluster.getRabbitMQ2().getHostIp(), cluster.getRabbitMQ2().getPort()),
+            new Address(cluster.getRabbitMQ3().getHostIp(), cluster.getRabbitMQ3().getPort()));
+        try (Connection connection = connectionFactory1.newConnection(addresses);
+             Channel channel = connection.createChannel();
+             Connection connection2 = connectionFactory2.newConnection(addresses);
+             Channel channel2 = connection2.createChannel()) {
+
+            channel.exchangeDeclare(EXCHANGE_NAME, "direct", DURABLE);
+            channel2.queueDeclare(QUEUE, DURABLE, !EXCLUSIVE, !AUTO_DELETE, ImmutableMap.of()).getQueue();
+            channel2.queueBind(QUEUE, EXCHANGE_NAME, ROUTING_KEY);
+
+            for (int i = 0; i < MESSAGE_COUNT; i++) {
+                if (i == MESSAGE_COUNT / 2) {
+                    cluster.getRabbitMQ1().stop();
+                }
+                channel.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPS, String.valueOf(i).getBytes(StandardCharsets.UTF_8));
+            }
+
+            AtomicInteger counter = new AtomicInteger(0);
+            channel2.basicConsume(QUEUE, callbackWithCrash(result, channel2, counter, cluster.getRabbitMQ1()));
+            Thread.sleep(ONE_MINUTE);
+        }
+
+        assertThat(result)
+            .containsOnlyElementsOf(
+                IntStream.range(0, MESSAGE_COUNT)
+                    .boxed()
+                    .collect(Guavate.toImmutableList()));
+    }
+
+    private DefaultConsumer callback(ConcurrentLinkedDeque<Integer> result, Channel channel2) {
```

Comment:
Why can't we use your awesome inmemoryConsumer stuff, that does the exact same job.

Reply:
s/channel2/channel/